### PR TITLE
support prop value for compose classes key

### PR DESCRIPTION
**Pull request by jakerichan**
This allows us to pass a prop value as a status and have it return only one class
i.e.
`this.composeClasses('ColorPicker', {'size-@': props.size})`
Will result in a status class of:
`ColorPicker-is-size-small`

As opposed to listing all possibilities as such:
`this.composeClasses('ColorPicker', {'small': size === 'small', 'normal': size === 'normal', 'flex': size === 'flex', 'round': size === 'round' 'selected': selected})`

### CODE REVIEW

File: utils/seed-classes.js [hunk 1]
```diff
@@ -24,7 +25,13 @@ export default function(base, statuses = {}, classes = {}) {
...
-    arr.push('-' + appendStatus(sx[s], s));
+    var status = templatePattern.test(s) ? appendStatus(true, useTemplate(s, sx[s])) : appendStatus(sx[s], s);
+    arr.push('-' + status);
   }
   return arr;
 }
+
+function useTemplate(template, value) {
+  template = template.replace(templatePattern, '${value}');
+  return eval('`' + template + '`');
```
Comment (camshaft) [inline utils/seed-classes.js]: no. no. no.

**Comment by camshaft** [comment]
you should use https://github.com/lang-js/interpolate instead. here's an example usage https://github.com/lang-js/interpolate/blob/master/test/index.js#L20-L22

**Comment by camshaft** [comment]
you can change the patterns as well https://github.com/lang-js/interpolate/blob/master/test/index.js#L20-L22

**Comment by devqzzr** [comment]
:thumbsup:

**Comment by mndvns** [comment]
who in the crap is devqzzr

**Pull request merged**